pkg/sitewise/api/list_associated_assets.go

```diff
@@ -14,6 +14,33 @@ import (
 
 func ListAssociatedAssets(ctx context.Context, client client.SitewiseClient, query models.ListAssociatedAssetsQuery) (*framer.AssociatedAssets, error) {
 
+	var assetId *string = util.GetAssetId(query.BaseQuery)
+
+	// Recursively load children
+	if query.LoadAllChildren {
+		asset, err := client.DescribeAsset(&iotsitewise.DescribeAssetInput{
+			AssetId: assetId,
+		})
+
+		if err != nil {
+			return nil, err
+		}
+
+		var results []*iotsitewise.AssociatedAssetsSummary
+
+		for _, h := range asset.AssetHierarchies {
+			resp, _ := client.ListAssociatedAssets(&iotsitewise.ListAssociatedAssetsInput{
```

Comment:
1.  you should handle error
2. the response may be paginated.

Reply:
This should be resolved, I also noticed I missed the maxResults and ctx parameters, so that's been added as well